tests/tel2puml/find_unique_graphs/test_otel_data_holder.py

```diff
@@ -29,55 +29,62 @@ def test_sql_data_holder_init(
+        assert holder.max_timestamp == 0
+        # Test attributes set correctly
+        assert holder.batch_size == 10
+        assert holder.node_models_to_save == []
+        assert holder.node_relationships_to_save == []
+
+    @staticmethod
+    def test_sql_data_holder_enter_method(
+        mock_sql_config: SQLDataHolderConfig,
+    ) -> None:
+        """Tests the __enter__ method using a context manager."""
+
         with patch.object(
             SQLDataHolder, "create_db_tables"
         ) as mock_create_db_tables:
-            holder = SQLDataHolder(mock_sql_config)
-            # Test super init method
-            assert (
-                holder.min_timestamp == 999999999999999999999999999999999999999
-            )
-            assert holder.max_timestamp == 0
-            # Test attributes set correctly
-            assert holder.batch_size == 10
-            assert holder.node_models_to_save == []
-            assert holder.node_relationships_to_save == []
-            assert isinstance(holder.engine, Engine)
-            assert holder.engine.url.drivername == "sqlite"
-            assert holder.engine.url.database == ":memory:"
-            assert isinstance(holder.session, Session)
-            assert isinstance(holder.base, Base)
-            # Test create_db_tables method is called
-            mock_create_db_tables.assert_called_once()
+            with SQLDataHolder(mock_sql_config) as holder:
+                assert isinstance(holder.engine, Engine)
+                assert holder.engine.url.drivername == "sqlite"
+                assert holder.engine.url.database == ":memory:"
+                assert isinstance(holder.session, Session)
+                assert isinstance(holder.base, Base)
+                # Test create_db_tables method is called
+                mock_create_db_tables.assert_called_once()
 
     @staticmethod
     def test_create_db_tables(mock_sql_config: SQLDataHolderConfig) -> None:
         """Tests the create_db_tables method."""
 
-        holder = SQLDataHolder(mock_sql_config)
-        holder.create_db_tables()
-        inspector = inspect(holder.engine)
-        # Test column names in nodes table
-        columns = inspector.get_columns("nodes")
-        column_names = [column["name"] for column in columns]
-        expected_column_names = [
-            "id",
-            "job_name",
-            "job_id",
-            "event_type",
-            "event_id",
-            "start_timestamp",
-            "end_timestamp",
-            "application_name",
-            "parent_event_id",
-        ]
-        assert column_names == expected_column_names
-
-        # Test column names in node association table
-        columns = inspector.get_columns("NODE_ASSOCIATION")
-        column_names = [column["name"] for column in columns]
-
-        assert column_names == ["parent_id", "child_id"]
+        with SQLDataHolder(mock_sql_config) as holder:
```

Comment:
I would revert this back re my comment above about not using context manager if we don't need to